### ISSUE TITLE
security: CVE-2023-49569 - go-git/v5

### ISSUE THREAD

**Issue by mfreeman451** [argoproj/argo-cd#16822]
Checklist:

* [x] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `argocd version`.

**Describe the bug**

quay.io/argoproj/argocd:v2.9.3 and other images appear to be vulnerable to several CVEs, including one that is a `CRITICAL` and a few that our opensource scanner has marked as 'HIGH'.

```json
      {
           "Target": "usr/local/bin/argocd",
           "Class": "lang-pkgs",
           "Type": "gobinary",
           "Vulnerabilities": [
             {
               "VulnerabilityID": "CVE-2023-49569",
               "PkgName": "github.com/go-git/go-git/v5",
               "InstalledVersion": "v5.8.1",
               "FixedVersion": "5.11.0",
               "Status": "fixed",
```

**To Reproduce**

```bash
docker pull quay.io/argoproj/argocd:v2.9.3
```

```bash
trivy image quay.io/argoproj/argocd:v2.9.3
```

**Expected behavior**

**Screenshots**

<img width="1268" alt="Screenshot 2024-01-10 at 10 28 35 PM" src="https://github.com/argoproj/argo-cd/assets/1821930/5cc2a02c-e552-484f-a63f-7b837c61401d">

**Version**

```shell
❯ argocd version
argocd: v2.9.3+6eba5be.dirty
  BuildDate: 2023-12-02T00:37:11Z
  GitCommit: 6eba5be864b7e031871ed7698f5233336dfe75c7
  GitTreeState: dirty
  GoVersion: go1.21.4
  Compiler: gc
  Platform: darwin/arm64
argocd-server: v2.9.3+6eba5be
```

**Logs**

```
Paste any relevant application logs here.
```

**Comment by SudhamshBachu** [comment]
Does any of the ArgoCD version address this Vuln ?

**Comment by crenshaw-dev** [comment]
Not afaik. Could you open a PR to bump the dependency?

**Comment by blakepettersson** [comment]
Should be fixed with #16711

**Comment by blakepettersson** [comment]
Closing issue, since #16711 has been merged.
